env0/utils.go

```diff
@@ -117,29 +117,38 @@ func readResourceData(i interface{}, d *schema.ResourceData) error {
 	return nil
 }
 
+// Returns the field name or skip.
+func getFieldName(field reflect.StructField) (string, bool) {
```

Comment:
As discussed, I've tried moving common code to functions.
Added some more comments. It's still very complex, but I think it's much more readable now.